In [2]:
import pandas as pd

In [3]:
#file = 'Interactive_Map_Data.csv'
#this is from the NYC Open Data

In [4]:
# previous encoding 
#df = pd.read_csv(file, encoding="ISO-8859-1")

In [19]:
file = 'nyc_movies.csv'
df = pd.read_csv(file, encoding='UTF-8')

In [20]:
df.head()

Film  Year               URL Encoded name  \
0  *batteries not included  1987  %2Abatteries%20not%20included   
1             12 Angry Men  1957               12%20Angry%20Men   
2           13 Going on 30  2004           13%20Going%20on%2030   
3               15 Minutes  2001                   15%20Minutes   
4                25th Hour  2002                    25th%20Hour   

           Image File Name                         Agency Credit  \
0  batteriesnotincluded_pf                 Courtesy of Photofest   
1            12AngryMen_pf                 Courtesy of Photofest   
2           13Goingon30_ec  Courtesy of Everett Collection, Inc.   
3                      NaN                                   NaN   
4             25thHour1_pf                 Courtesy of Photofest   

  Artist Credit Director/Filmmaker Name         Director/Filmmaker IMDB Link  \
0   Directed by         Matthew Robbins      http://imdb.com/name/nm0730422/   
1   Directed by            Sidney Lumet      http://imdb.com/name/nm0001486/   
2   Directed by             Gary Winick      http://imdb.com/name/nm0935095/   
3   Directed by           John Herzfeld      http://imdb.com/name/nm0381273/   
4   Directed by               Spike Lee  http://www.imdb.com/name/nm0000490/   

                               Location Display Text   LATITUDE  LONGITUDE  \
0            E. 5th St.<br>East Village<br>Manhattan  40.722445 -73.978651   
1  New York County Courthouse<br>40 Foley Square<...  40.713700 -74.007900   
2  W. 47th St. and Seventh Ave.<br>Times Square<b...  40.759220 -73.984621   
3  E. 60-66th St. and Madison Ave.<br>Upper East ...  40.766100 -73.969600   
4              World Trade Center<br>Lower Manhattan  40.711793 -74.012328   

     Borough     Neighborhood Scene Type Media  \
0  Manhattan     East Village        NaN  Film   
1  Manhattan  Lower Manhattan        NaN  Film   
2  Manhattan     Times Square        NaN  Film   
3  Manhattan  Upper East Side     Chases  Film   
4  Manhattan  Lower Manhattan        NaN  Film   

                              IMDB LINK  \
0      http://imdb.com/title/tt0092494/   
1      http://imdb.com/title/tt0050083/   
2  http://www.imdb.com/title/tt0337563/   
3  http://www.imdb.com/title/tt0179626/   
4  http://www.imdb.com/title/tt0307901/   

               Client or book location indicator Notes Book Image Book Page  \
0                                            NaN   NaN          Y       190   
1   New York County\nCourthouse on Foley Square.   NaN          Y        40   
2  47th St. and 7th Ave. Times Square\nManhattan   NaN          Y       239   
3                            60-66th and Madison   NaN          Y        82   
4                             World Trade Center   NaN          Y        93   

  Display? IMAGE OF LOCATION  ratings  
0      NaN                 Y      6.7  
1      NaN                 Y      9.0  
2      NaN                 Y      6.2  
3      NaN                 N      6.1  
4      NaN                 Y      7.6

In [7]:
df['ratings'] = 0

In [8]:
df

Film  Year                URL Encoded name  \
0    *batteries not included  1987   %2Abatteries%20not%20included   
1               12 Angry Men  1957                12%20Angry%20Men   
2             13 Going on 30  2004            13%20Going%20on%2030   
3                 15 Minutes  2001                    15%20Minutes   
4                  25th Hour  2002                     25th%20Hour   
..                       ...   ...                             ...   
228             Working Girl  1988                  Working%20Girl   
229     You're a Big Boy Now  1966  You%27re%20a%20Big%20Boy%20Now   
230     You're a Big Boy Now  1966  You%27re%20a%20Big%20Boy%20Now   
231          You've Got Mail  1998           You%27ve%20Got%20Mail   
232          You've Got Mail  1998           You%27ve%20Got%20Mail   

             Image File Name                         Agency Credit  \
0    batteriesnotincluded_pf                 Courtesy of Photofest   
1              12AngryMen_pf                 Courtesy of Photofest   
2             13Goingon30_ec  Courtesy of Everett Collection, Inc.   
3                        NaN                                   NaN   
4               25thHour1_pf                 Courtesy of Photofest   
..                       ...                                   ...   
228           WorkingGirl_pf                 Courtesy of Photofest   
229       YoureaBigBoyNow_pf                 Courtesy of Photofest   
230      YoureaBigBoyNow2_pf                 Courtesy of Photofest   
231         YouveGotMail1_pf                 Courtesy of Photofest   
232         YouveGotMail2_pf                 Courtesy of Photofest   

    Artist Credit Director/Filmmaker Name  \
0     Directed by         Matthew Robbins   
1     Directed by            Sidney Lumet   
2     Directed by             Gary Winick   
3     Directed by           John Herzfeld   
4     Directed by               Spike Lee   
..            ...                     ...   
228   Directed by            Mike Nichols   
229   Directed by    Francis Ford Coppola   
230   Directed by    Francis Ford Coppola   
231   Directed by             Nora Ephron   
232   Directed by             Nora Ephron   

            Director/Filmmaker IMDB Link  \
0        http://imdb.com/name/nm0730422/   
1        http://imdb.com/name/nm0001486/   
2        http://imdb.com/name/nm0935095/   
3        http://imdb.com/name/nm0381273/   
4    http://www.imdb.com/name/nm0000490/   
..                                   ...   
228      http://imdb.com/name/nm0001566/   
229      http://imdb.com/name/nm0000338/   
230      http://imdb.com/name/nm0000338/   
231      http://imdb.com/name/nm0001188/   
232      http://imdb.com/name/nm0001188/   

                                 Location Display Text   LATITUDE  ...  \
0              E. 5th St.<br>East Village<br>Manhattan  40.722445  ...   
1    New York County Courthouse<br>40 Foley Square<...  40.713700  ...   
2    W. 47th St. and Seventh Ave.<br>Times Square<b...  40.759220  ...   
3    E. 60-66th St. and Madison Ave.<br>Upper East ...  40.766100  ...   
4                World Trade Center<br>Lower Manhattan  40.711793  ...   
..                                                 ...        ...  ...   
228             Staten Island Ferry<br>New York Harbor  40.682330  ...   
229      Steeplechase Park<br>Coney Island<br>Brooklyn  40.574726  ...   
230  New York Public Library<br>Fifth Ave. and 41st...  40.752865  ...   
231  Verdi Square<br>W. 73rd St.  & Broadway<br>Man...  40.779258  ...   
232  91st Street Community Garden<br>Riverside Park...  40.792824  ...   

     Scene Type Media                             IMDB LINK  \
0           NaN  Film      http://imdb.com/title/tt0092494/   
1           NaN  Film      http://imdb.com/title/tt0050083/   
2           NaN  Film  http://www.imdb.com/title/tt0337563/   
3        Chases  Film  http://www.imdb.com/title/tt0179626/   
4           NaN  Film  http://www.imdb.com/title/tt0307901/   
..   

In [21]:
import requests
import json
import configparser
from pprint import pprint


In [23]:
url = "http://www.omdbapi.com/?t="
api_key = "&apikey=" + api_key1

In [10]:
df['Film'][0]

'*batteries not included'

In [11]:
movie = df['Film'][4]
print(movie)

25th Hour


In [24]:
response = requests.get(f"http://www.omdbapi.com/?apikey=72bc447a&t={movie}")

In [13]:
# Converting the response to JSON, and printing the result.
data = response.json()
pprint(data)

{'Actors': 'Edward Norton, Philip Seymour Hoffman, Barry Pepper, Rosario '
           'Dawson',
 'Awards': 'Nominated for 1 Golden Globe. Another 4 wins & 16 nominations.',
 'BoxOffice': '$13,084,595',
 'Country': 'USA',
 'DVD': 'N/A',
 'Director': 'Spike Lee',
 'Genre': 'Drama',
 'Language': 'English',
 'Metascore': '68',
 'Plot': 'Cornered by the DEA, convicted New York drug dealer Montgomery '
         'Brogan reevaluates his life in the 24 remaining hours before facing '
         'a seven-year jail term.',
 'Poster': 'https://m.media-amazon.com/images/M/MV5BNmE0YjdlYTktMTU4Ni00Mjk2LWI3NWMtM2RjNmFiOTk4YjYxL2ltYWdlXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg',
 'Production': '40 Acres &amp; A Mule Filmworks, Gamut Films, Industry '
               'Entertainment',
 'Rated': 'R',
 'Ratings': [{'Source': 'Internet Movie Database', 'Value': '7.6/10'},
             {'Source': 'Rotten Tomatoes', 'Value': '78%'},
             {'Source': 'Metacritic', 'Value': '68/100'}],
 'Released': '10 Jan 

In [44]:
data['Awards']

'Nominated for 3 Oscars. Another 17 wins & 10 nominations.'

In [ ]:
df = 0.0

In [20]:
data['imdbRating']

'7.6'

In [22]:
next(df.iterrows())

(0,
 Film                                                 *batteries not included
 Year                                                                    1987
 URL Encoded name                               %2Abatteries%20not%20included
 Image File Name                                      batteriesnotincluded_pf
 Agency Credit                                          Courtesy of Photofest
 Artist Credit                                                    Directed by
 Director/Filmmaker Name                                      Matthew Robbins
 Director/Filmmaker IMDB Link                 http://imdb.com/name/nm0730422/
 Location Display Text                E. 5th St.<br>East Village<br>Manhattan
 LATITUDE                                                             40.7224
 LONGITUDE                                                           -73.9787
 Borough                                                            Manhattan
 Neighborhood                                               

In [25]:
from time import sleep


In [26]:
import numpy as np

In [29]:
df['test'] = np.nan

In [31]:
del df['test']

In [27]:
df['Awards'] = np.nan
df['Box Office'] = np.nan
df['Director'] = np.nan
df['Genre'] = np.nan
df['Language'] = np.nan
df['Plot'] = np.nan
df['Poster'] = np.nan
df['Rated'] = np.nan
df['Runtime'] = np.nan
df['imdbID'] = np.nan
df['imdbVotes'] = np.nan

In [17]:
pd.set_option('display.max_columns', None)


{'Actors': 'Edward Norton, Philip Seymour Hoffman, Barry Pepper, Rosario '
           'Dawson',
 'Awards': 'Nominated for 1 Golden Globe. Another 4 wins & 16 nominations.',
 'BoxOffice': '$13,084,595',
 'Country': 'USA',
 'DVD': 'N/A',
 'Director': 'Spike Lee',
 'Genre': 'Drama',
 'Language': 'English',
 'Metascore': '68',
 'Plot': 'Cornered by the DEA, convicted New York drug dealer Montgomery '
         'Brogan reevaluates his life in the 24 remaining hours before facing '
         'a seven-year jail term.',
 'Poster': 'https://m.media-amazon.com/images/M/MV5BNmE0YjdlYTktMTU4Ni00Mjk2LWI3NWMtM2RjNmFiOTk4YjYxL2ltYWdlXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg',
 'Production': '40 Acres &amp; A Mule Filmworks, Gamut Films, Industry '
               'Entertainment',
 'Rated': 'R',
 'Ratings': [{'Source': 'Internet Movie Database', 'Value': '7.6/10'},
             {'Source': 'Rotten Tomatoes', 'Value': '78%'},
             {'Source': 'Metacritic', 'Value': '68/100'}],
 'Released': '10 Jan 2003',
 'Response': 'True',
 'Runtime': '135 min',
 'Title': '25th Hour',
 'Type': 'movie',
 'Website': 'N/A',
 'Writer': 'David Benioff (novel), David Benioff (screenplay)',
 'Year': '2002',
 'imdbID': 'tt0307901',
 'imdbRating': '7.6',
 'imdbVotes': '169,256'}

In [56]:
df['Awards'] = np.nan
df['Box Office'] = np.nan
df['Director'] = np.nan
df['Genre'] = np.nan
df['Plot'] = np.nan
df['Poster'] = np.nan
df['Rated'] = np.nan
df['Runtime'] = np.nan
df['imdbID'] = np.nan
df['imdbVotes'] = np.nan

print(data['Awards'])
print(data['BoxOffice'])
print(data['Director'])
print(data['Genre'])
print(data['Plot'])
print(data['Poster'])
print(data['Rated'])
print(data['Runtime'])
print(data['imdbID'])
print(data['imdbVotes'])

Nominated for 3 Oscars. Another 17 wins & 10 nominations.
N/A
Sidney Lumet
Crime, Drama
A jury holdout attempts to prevent a miscarriage of justice by forcing his colleagues to reconsider the evidence.
https://m.media-amazon.com/images/M/MV5BMWU4N2FjNzYtNTVkNC00NzQ0LTg0MjAtYTJlMjFhNGUxZDFmXkEyXkFqcGdeQXVyNjc1NTYyMjg@._V1_SX300.jpg
Approved
96 min
tt0050083
682,907


In [28]:
df = df.copy()

In [29]:
df.head(1)

Film  Year               URL Encoded name  \
0  *batteries not included  1987  %2Abatteries%20not%20included   

           Image File Name          Agency Credit Artist Credit  \
0  batteriesnotincluded_pf  Courtesy of Photofest   Directed by   

  Director/Filmmaker Name     Director/Filmmaker IMDB Link  \
0         Matthew Robbins  http://imdb.com/name/nm0730422/   

                     Location Display Text   LATITUDE  LONGITUDE    Borough  \
0  E. 5th St.<br>East Village<br>Manhattan  40.722445 -73.978651  Manhattan   

   Neighborhood Scene Type Media                         IMDB LINK  \
0  East Village        NaN  Film  http://imdb.com/title/tt0092494/   

  Client or book location indicator Notes Book Image Book Page Display?  \
0                               NaN   NaN          Y       190      NaN   

  IMAGE OF LOCATION  ratings  Awards  Box Office  Director  Genre  Language  \
0                 Y      6.7     NaN         NaN       NaN    NaN       NaN   

   Plot  Poster  Rated  Runtime  imdbID  imdbVotes  
0   NaN     NaN    NaN      NaN     NaN        NaN

In [30]:
# add a timeout here:
# df['Awards'] = np.nan
# df['Box Office'] = np.nan
# df['Director'] = np.nan
# df['Genre'] = np.nan
# df['Language'] = np.nan
# df['Plot'] = np.nan
# df['Poster'] = np.nan
# df['Rated'] = np.nan
# df['Runtime'] = np.nan
# df['Year'] = np.nan
# df['imdbID'] = np.nan
# df['imdbVotes'] = np.nan

# print(data['Awards'])
# print(data['BoxOffice'])
# print(data['Director'])
# print(data['Genre'])
# print(data['Language'])
# print(data['Plot'])
# print(data['Poster'])
# print(data['Rated'])
# print(data['Runtime'])
# print(data['imdbId'])
# print(data['ImdbVotes'])

for index, row in df.iterrows():
    try:
        movie = df['Film'][index]
        response = requests.get(f"http://www.omdbapi.com/?apikey=72bc447a&t={movie}")
        data = response.json()
        try: 
            df['Box Office'][index] = data['BoxOffice']
        except: 
            print(f"none Box Office {index}")
        try: 
            df['Director'][index] = data['Director']
        except: 
            print(f"none Director {index}")
        try:
            df['Genre'][index] = data["Genre"]
        except:
            print(f"none Genre {index}")
        try:
            df['Plot'][index] = data["Plot"]
        except:
            print(f"none Plot {index}")
        try:
            df['Rated'][index] = data['Rated']
        except:
            print(f"none Rated {index}")
        try:
            df['Ratings'][index] = data['Ratings']
        except:
            print(f"none Runtime {index}")
        try:
            df['imdbID'][index] = data['imdbID']
        except: 
            print(f"none imdbId {index}")
        try:
            df['imdbVotes'][index] = data['imdbVotes']
        except:
            print(f"none imdbVotes {index}")
#         df['ratings'][index] = data['Ratings'][0]['Value']
      
# print(foo)
    except:
        print('nope')
    sleep(0.5)  

<ipython-input-30-bb1a88b9607a>:33: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Awards'][index] = data['Awards']
/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexing.py:670: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  iloc._setitem_with_indexer(indexer, value)
<ipython-input-30-bb1a88b9607a>:37: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Box Office'][index] = data['BoxOffice']
<ipython-input-30-bb1a88b9607a>:4

none Box Office 30
none Awards50
none Box Office 50
none Director 50
none Genre 50
none Plot 50
none Poster 50
none Rated 50
none Runtime 50
none imdbId 50
none imdbVotes 50
none Box Office 93
none Box Office 94
none Box Office 105
none Awards115
none Box Office 115
none Director 115
none Genre 115
none Plot 115
none Poster 115
none Rated 115
none Runtime 115
none imdbId 115
none imdbVotes 115
none Box Office 140
none Awards185
none Box Office 185
none Director 185
none Genre 185
none Plot 185
none Poster 185
none Rated 185
none Runtime 185
none imdbId 185
none imdbVotes 185
none Box Office 202
none Awards222
none Box Office 222
none Director 222
none Genre 222
none Plot 222
none Poster 222
none Rated 222
none Runtime 222
none imdbId 222
none imdbVotes 222


In [46]:
df.iloc[115]

Film                                                              My Dinner with Andre
Year                                                                              1991
URL Encoded name                                          My%20Dinner%20with%20Andr%E9
Image File Name                                                   MyDinnerWithAndre_pf
Agency Credit                                                    Courtesy of Photofest
Artist Credit                                                              Directed by
Director/Filmmaker Name                                                    Louis Malle
Director/Filmmaker IMDB Link                           http://imdb.com/name/nm0001501/
Location Display Text                                                          Subway 
LATITUDE                                                                       40.7186
LONGITUDE                                                                     -73.9884
Borough                                    

In [39]:
df['Film'][222] = 'When Harry Met Sally'

<ipython-input-39-3a82fc810af5>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Film'][222] = 'When Harry Met Sally'


In [ ]:
df['Fil']

In [43]:
        movie = df['Film'][50]
print(movie)

Downtown 81


In [42]:
        movie = df['Film'][115]
        response = requests.get(f"http://www.omdbapi.com/?apikey=72bc447a&t={movie}")
        data = response.json()
        try: 
            df['Awards'][115] = data['Awards']
        except: 
            print(f"none Awards{115}")
        try: 
            df['Box Office'][115] = data['BoxOffice']
        except: 
            print(f"none Box Office {115}")
        try: 
            df['Director'][115] = data['Director']
        except: 
            print(f"none Director {115}")
        try:
            df['Genre'][115] = data["Genre"]
        except:
            print(f"none Genre {115}")
        try:
            df['Plot'][115] = data["Plot"]
        except:
            print(f"none Plot {115}")
        try: 
            df['Poster'][115] = data['Poster']
        except:
            print(f"none Poster {115}")
        try:
            df['Rated'][115] = data['Rated']
        except:
            print(f"none Rated {115}")
        try:
            df['Runtime'][115] = data['Runtime']
        except:
            print(f"none Runtime {115}")
        try:
            df['imdbID'][115] = data['imdbID']
        except: 
            print(f"none imdbId {115}")
        try:
            df['imdbVotes'][115] = data['imdbVotes']
        except:
            print(f"none imdbVotes {115}")

<ipython-input-42-64582ab1b71a>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Awards'][115] = data['Awards']
<ipython-input-42-64582ab1b71a>:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Box Office'][115] = data['BoxOffice']
<ipython-input-42-64582ab1b71a>:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Director'][115] = data['Director']
<ipython-input-42-64582ab1b71a>:17: SettingWithCopyWarning: 
A value is trying 

In [78]:
df.iloc[222]

Film                                                              When Harry Met Sally
Year                                                                               NaN
URL Encoded name                                         When%20Harry%20Met%20Sally%85
Image File Name                                                  WhenHarryMetSally9_pf
Agency Credit                                     Courtesy of Everett Collection, Inc.
Artist Credit                                                              Directed by
Director/Filmmaker Name                                                     Rob Reiner
Director/Filmmaker IMDB Link                       http://www.imdb.com/name/nm0001661/
Location Display Text                   Katz's Deli<br>205 E. Houston St.<br>Manhattan
LATITUDE                                                                       40.7223
LONGITUDE                                                                     -73.9874
Borough                                    

In [71]:
df.iloc[115,0] = "My Dinner With Andre"

In [72]:
df.iloc[115,0]

'My Dinner With Andre'

In [73]:
df.iloc[222, 0]

'When Harry Met SallyÉ'

In [74]:
df.iloc[222, 0] = 'When Harry Met Sally'

In [68]:
df.iloc[115,]

Film                                             My Dinner with AndrŽ
Year                                                              NaN
URL Encoded name                         My%20Dinner%20with%20Andr%E9
Image File Name                                  MyDinnerWithAndre_pf
Agency Credit                                   Courtesy of Photofest
Artist Credit                                             Directed by
Director/Filmmaker Name                                   Louis Malle
Director/Filmmaker IMDB Link          http://imdb.com/name/nm0001501/
Location Display Text                                         Subway 
LATITUDE                                                      40.7186
LONGITUDE                                                    -73.9884
Borough                                                     Manhattan
Neighborhood                                          Lower East Side
Scene Type                                                 The Subway
Media               

In [75]:
len(df['ratings'][0])

3

In [74]:
df['ratings'].tolist()

['6.7',
 '9.0',
 '6.2',
 '6.1',
 '7.6',
 '7.6',
 '6.5',
 '7.4',
 '6.0',
 '7.7',
 '6.6',
 '7.6',
 '7.1',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '8.0',
 '5.7',
 '7.7',
 '7.0',
 '7.0',
 '6.9',
 '6.1',
 '7.3',
 '6.7',
 '7.6',
 '7.6',
 '7.6',
 '7.4',
 '6.8',
 '6.7',
 '7.2',
 '5.2',
 '6.9',
 '6.5',
 '6.5',
 '6.6',
 '5.7',
 '7.0',
 '6.9',
 '6.5',
 '6.5',
 '6.4',
 '6.3',
 '6.0',
 '6.0',
 '7.6',
 '8.0',
 '8.0',
 0,
 0,
 '8.3/10',
 '6.6/10',
 '6.6/10',
 '7.7/10',
 '7.3/10',
 '6.7/10',
 '7.8/10',
 '7.8/10',
 '6.5/10',
 '6.5/10',
 '6.5/10',
 '6.5/10',
 '6.5/10',
 '6.5/10',
 '6.5/10',
 '6.5/10',
 '6.4/10',
 '8.7/10',
 '6.2/10',
 '5.7/10',
 '7.6/10',
 '7.9/10',
 '7.9/10',
 '7.9/10',
 '6.9/10',
 '6.6/10',
 '5.8/10',
 '6.9/10',
 '6.6/10',
 '7.6/10',
 '6.4/10',
 '6.9/10',
 '5.7/10',
 '6.6/10',
 '6.6/10',
 '7.1/10',
 '6.6/10',
 '7.2/10',
 '6.4/10',
 '7.1/10',
 '7.1/10',
 '7.1/10',
 '7.1/10',
 '7.1/10',
 '7.8/10',
 '7.4/10',
 '7.5/10',
 '7.5/10',
 '6.0/10',
 '7.3/10',
 '7.7/10',
 '7.0/10',
 '7.9/10',
 '7.7/10'

In [80]:
df

Film  Year                URL Encoded name  \
0    *batteries not included  1987   %2Abatteries%20not%20included   
1               12 Angry Men  1957                12%20Angry%20Men   
2             13 Going on 30  2004            13%20Going%20on%2030   
3                 15 Minutes  2001                    15%20Minutes   
4                  25th Hour  2002                     25th%20Hour   
..                       ...   ...                             ...   
228             Working Girl  1988                  Working%20Girl   
229     You're a Big Boy Now  1966  You%27re%20a%20Big%20Boy%20Now   
230     You're a Big Boy Now  1966  You%27re%20a%20Big%20Boy%20Now   
231          You've Got Mail  1998           You%27ve%20Got%20Mail   
232          You've Got Mail  1998           You%27ve%20Got%20Mail   

             Image File Name                         Agency Credit  \
0    batteriesnotincluded_pf                 Courtesy of Photofest   
1              12AngryMen_pf                 Courtesy of Photofest   
2             13Goingon30_ec  Courtesy of Everett Collection, Inc.   
3                        NaN                                   NaN   
4               25thHour1_pf                 Courtesy of Photofest   
..                       ...                                   ...   
228           WorkingGirl_pf                 Courtesy of Photofest   
229       YoureaBigBoyNow_pf                 Courtesy of Photofest   
230      YoureaBigBoyNow2_pf                 Courtesy of Photofest   
231         YouveGotMail1_pf                 Courtesy of Photofest   
232         YouveGotMail2_pf                 Courtesy of Photofest   

    Artist Credit Director/Filmmaker Name  \
0     Directed by         Matthew Robbins   
1     Directed by            Sidney Lumet   
2     Directed by             Gary Winick   
3     Directed by           John Herzfeld   
4     Directed by               Spike Lee   
..            ...                     ...   
228   Directed by            Mike Nichols   
229   Directed by    Francis Ford Coppola   
230   Directed by    Francis Ford Coppola   
231   Directed by             Nora Ephron   
232   Directed by             Nora Ephron   

            Director/Filmmaker IMDB Link  \
0        http://imdb.com/name/nm0730422/   
1        http://imdb.com/name/nm0001486/   
2        http://imdb.com/name/nm0935095/   
3        http://imdb.com/name/nm0381273/   
4    http://www.imdb.com/name/nm0000490/   
..                                   ...   
228      http://imdb.com/name/nm0001566/   
229      http://imdb.com/name/nm0000338/   
230      http://imdb.com/name/nm0000338/   
231      http://imdb.com/name/nm0001188/   
232      http://imdb.com/name/nm0001188/   

                                 Location Display Text   LATITUDE  ...  \
0              E. 5th St.<br>East Village<br>Manhattan  40.722445  ...   
1    New York County Courthouse<br>40 Foley Square<...  40.713700  ...   
2    W. 47th St. and Seventh Ave.<br>Times Square<b...  40.759220  ...   
3    E. 60-66th St. and Madison Ave.<br>Upper East ...  40.766100  ...   
4                World Trade Center<br>Lower Manhattan  40.711793  ...   
..                                                 ...        ...  ...   
228             Staten Island Ferry<br>New York Harbor  40.682330  ...   
229      Steeplechase Park<br>Coney Island<br>Brooklyn  40.574726  ...   
230  New York Public Library<br>Fifth Ave. and 41st...  40.752865  ...   
231  Verdi Square<br>W. 73rd St.  & Broadway<br>Man...  40.779258  ...   
232  91st Street Community Garden<br>Riverside Park...  40.792824  ...   

     Scene Type Media                             IMDB LINK  \
0           NaN  Film      http://imdb.com/title/tt0092494/   
1           NaN  Film      http://imdb.com/title/tt0050083/   
2           NaN  Film  http://www.imdb.com/title/tt0337563/   
3        Chases  Film  http://www.imdb.com/title/tt0179626/   
4           NaN  Film  http://www.imdb.com/title/tt0307901/   
..   

In [78]:
df['ratings'] = df['ratings'].astype('float') 

In [47]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 231 entries, 0 to 230
Data columns (total 34 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Film                               231 non-null    object 
 1   Year                               231 non-null    object 
 2   URL Encoded name                   231 non-null    object 
 3   Image File Name                    180 non-null    object 
 4   Agency Credit                      181 non-null    object 
 5   Artist Credit                      231 non-null    object 
 6   Director/Filmmaker Name            231 non-null    object 
 7   Director/Filmmaker IMDB Link       231 non-null    object 
 8   Location Display Text              231 non-null    object 
 9   LATITUDE                           231 non-null    float64
 10  LONGITUDE                          231 non-null    float64
 11  Borough                            231 non-null    object 

In [76]:
for index, row in df.iterrows():
    try:
        if len(df['ratings'][index]) > 3:
            df['ratings'][index] = df['ratings'][index][:-3]
    except:
        print('nope')

nope
nope
nope
nope
nope
nope


<ipython-input-76-931a784ac15a>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['ratings'][index] = df['ratings'][index][:-3]


In [81]:
df['ratings'].tolist()

[6.7,
 9.0,
 6.2,
 6.1,
 7.6,
 7.6,
 6.5,
 7.4,
 6.0,
 7.7,
 6.6,
 7.6,
 7.1,
 8.0,
 8.0,
 8.0,
 8.0,
 8.0,
 5.7,
 7.7,
 7.0,
 7.0,
 6.9,
 6.1,
 7.3,
 6.7,
 7.6,
 7.6,
 7.6,
 7.4,
 6.8,
 6.7,
 7.2,
 5.2,
 6.9,
 6.5,
 6.5,
 6.6,
 5.7,
 7.0,
 6.9,
 6.5,
 6.5,
 6.4,
 6.3,
 6.0,
 6.0,
 7.6,
 8.0,
 8.0,
 0.0,
 0.0,
 8.3,
 6.6,
 6.6,
 7.7,
 7.3,
 6.7,
 7.8,
 7.8,
 6.5,
 6.5,
 6.5,
 6.5,
 6.5,
 6.5,
 6.5,
 6.5,
 6.4,
 8.7,
 6.2,
 5.7,
 7.6,
 7.9,
 7.9,
 7.9,
 6.9,
 6.6,
 5.8,
 6.9,
 6.6,
 7.6,
 6.4,
 6.9,
 5.7,
 6.6,
 6.6,
 7.1,
 6.6,
 7.2,
 6.4,
 7.1,
 7.1,
 7.1,
 7.1,
 7.1,
 7.8,
 7.4,
 7.5,
 7.5,
 6.0,
 7.3,
 7.7,
 7.0,
 7.9,
 7.7,
 6.9,
 7.2,
 7.3,
 7.3,
 7.3,
 7.3,
 7.8,
 7.8,
 7.4,
 6.6,
 0.0,
 6.7,
 6.9,
 6.4,
 7.0,
 7.0,
 8.3,
 7.4,
 8.1,
 6.5,
 6.5,
 6.2,
 7.0,
 6.6,
 6.2,
 7.3,
 6.4,
 8.3,
 8.0,
 8.0,
 8.0,
 6.8,
 6.8,
 7.4,
 7.7,
 8.1,
 5.6,
 7.3,
 5.9,
 5.9,
 6.0,
 6.4,
 6.8,
 6.8,
 7.6,
 7.6,
 6.8,
 7.4,
 7.6,
 7.3,
 7.3,
 7.3,
 6.3,
 7.2,
 5.9,
 6.7,
 6.2,
 6.7,
 8.1,
 8.3,
 8.3

In [99]:
df.loc[df["ratings"] == 0.0, :]

Film    Year  \
50    Downtown 81 / New York Beat Movie    1981   
116                My Dinner with AndrŽ    1991   
186  The Late Show with David Letterman  1993-?   

                                    URL Encoded name       Image File Name  \
50   Downtown%2081%20%2F%20New%20York%20Beat%20Movie       Downtown81_2_pf   
116                     My%20Dinner%20with%20Andr%E9  MyDinnerWithAndre_pf   
186     The%20Late%20Show%20with%20David%20Letterman                   NaN   

             Agency Credit Artist Credit            Director/Filmmaker Name  \
50   Courtesy of Photofest   Directed by                      Edo Bertoglio   
116  Courtesy of Photofest   Directed by                        Louis Malle   
186                    NaN   Directed by  Jerry Foley,Randi Grossack,(more)   

                          Director/Filmmaker IMDB Link  \
50                     http://imdb.com/name/nm0078166/   
116                    http://imdb.com/name/nm0001501/   
186  http://imdb.com/name/nm0284159/,http://www.imd...   

                                 Location Display Text   LATITUDE  ...  \
50   The Empire State Building<br>Midtown<br>Manhattan  40.748346  ...   
116                                            Subway   40.718607  ...   
186  Ed Sullivan Theatre<br>1697 Broadway<br>Manhattan  40.763719  ...   

     Scene Type Media                             IMDB LINK  \
50          NaN  Film  http://www.imdb.com/title/tt0208993/   
116  The Subway  Film      http://imdb.com/title/tt0082783/   
186         NaN    TV  http://www.imdb.com/title/tt0106053/   

                     Client or book location indicator                 Notes  \
50   The Empire State building, 34th St and 5th Avenue                   NaN   
116                                            Subway   needs to be reversed   
186                                      1697 Broadway                   NaN   

    Book Image Book Page Display? IMAGE OF LOCATION ratings  
50           N       125      NaN                 Y     0.0  
116          Y       158      NaN                 Y     0.0  
186          Y       264      NaN                 N     0.0  

[3 rows x 23 columns]

In [91]:
df['ratings'][50]

0.0

In [97]:
df

Film  Year                URL Encoded name  \
0    *batteries not included  1987   %2Abatteries%20not%20included   
1               12 Angry Men  1957                12%20Angry%20Men   
2             13 Going on 30  2004            13%20Going%20on%2030   
3                 15 Minutes  2001                    15%20Minutes   
4                  25th Hour  2002                     25th%20Hour   
..                       ...   ...                             ...   
228             Working Girl  1988                  Working%20Girl   
229     You're a Big Boy Now  1966  You%27re%20a%20Big%20Boy%20Now   
230     You're a Big Boy Now  1966  You%27re%20a%20Big%20Boy%20Now   
231          You've Got Mail  1998           You%27ve%20Got%20Mail   
232          You've Got Mail  1998           You%27ve%20Got%20Mail   

             Image File Name                         Agency Credit  \
0    batteriesnotincluded_pf                 Courtesy of Photofest   
1              12AngryMen_pf                 Courtesy of Photofest   
2             13Goingon30_ec  Courtesy of Everett Collection, Inc.   
3                        NaN                                   NaN   
4               25thHour1_pf                 Courtesy of Photofest   
..                       ...                                   ...   
228           WorkingGirl_pf                 Courtesy of Photofest   
229       YoureaBigBoyNow_pf                 Courtesy of Photofest   
230      YoureaBigBoyNow2_pf                 Courtesy of Photofest   
231         YouveGotMail1_pf                 Courtesy of Photofest   
232         YouveGotMail2_pf                 Courtesy of Photofest   

    Artist Credit Director/Filmmaker Name  \
0     Directed by         Matthew Robbins   
1     Directed by            Sidney Lumet   
2     Directed by             Gary Winick   
3     Directed by           John Herzfeld   
4     Directed by               Spike Lee   
..            ...                     ...   
228   Directed by            Mike Nichols   
229   Directed by    Francis Ford Coppola   
230   Directed by    Francis Ford Coppola   
231   Directed by             Nora Ephron   
232   Directed by             Nora Ephron   

            Director/Filmmaker IMDB Link  \
0        http://imdb.com/name/nm0730422/   
1        http://imdb.com/name/nm0001486/   
2        http://imdb.com/name/nm0935095/   
3        http://imdb.com/name/nm0381273/   
4    http://www.imdb.com/name/nm0000490/   
..                                   ...   
228      http://imdb.com/name/nm0001566/   
229      http://imdb.com/name/nm0000338/   
230      http://imdb.com/name/nm0000338/   
231      http://imdb.com/name/nm0001188/   
232      http://imdb.com/name/nm0001188/   

                                 Location Display Text   LATITUDE  ...  \
0              E. 5th St.<br>East Village<br>Manhattan  40.722445  ...   
1    New York County Courthouse<br>40 Foley Square<...  40.713700  ...   
2    W. 47th St. and Seventh Ave.<br>Times Square<b...  40.759220  ...   
3    E. 60-66th St. and Madison Ave.<br>Upper East ...  40.766100  ...   
4                World Trade Center<br>Lower Manhattan  40.711793  ...   
..                                                 ...        ...  ...   
228             Staten Island Ferry<br>New York Harbor  40.682330  ...   
229      Steeplechase Park<br>Coney Island<br>Brooklyn  40.574726  ...   
230  New York Public Library<br>Fifth Ave. and 41st...  40.752865  ...   
231  Verdi Square<br>W. 73rd St.  & Broadway<br>Man...  40.779258  ...   
232  91st Street Community Garden<br>Riverside Park...  40.792824  ...   

     Scene Type Media                             IMDB LINK  \
0           NaN  Film      http://imdb.com/title/tt0092494/   
1           NaN  Film      http://imdb.com/title/tt0050083/   
2           NaN  Film  http://www.imdb.com/title/tt0337563/   
3        Chases  Film  http://www.imdb.com/title/tt0179626/   
4           NaN  Film  http://www.imdb.com/title/tt0307901/   
..   

In [50]:
df.describe()

LATITUDE   LONGITUDE     ratings  Language
count  231.000000  231.000000  231.000000     231.0
mean    40.739816  -73.979165    6.917316       0.0
std      0.052986    0.028234    1.117121       0.0
min     40.572232  -74.072721    0.000000       0.0
25%     40.713576  -73.996401    6.500000       0.0
50%     40.752589  -73.982100    7.000000       0.0
75%     40.772323  -73.970655    7.600000       0.0
max     40.859460  -73.841257    9.200000       0.0

In [82]:
file = 'nyc_movies_with_all_info.csv'
df3 = pd.read_csv(file, encoding='UTF-8')

In [56]:
pwd

'/Users/stanleytan/Desktop/bootcampspot/4-Project/2nd_project/second_step/movies'

In [96]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 212 entries, 0 to 230
Data columns (total 33 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Film                               212 non-null    object 
 1   Year                               212 non-null    float64
 2   URL Encoded name                   212 non-null    object 
 3   Image File Name                    212 non-null    object 
 4   Agency Credit                      212 non-null    object 
 5   Artist Credit                      212 non-null    object 
 6   Director/Filmmaker Name            212 non-null    object 
 7   Director/Filmmaker IMDB Link       212 non-null    object 
 8   Location Display Text              212 non-null    object 
 9   LATITUDE                           212 non-null    float64
 10  LONGITUDE                          212 non-null    float64
 11  Borough                            212 non-null    object 

In [49]:
df = df.fillna(0)

In [92]:
df3['ratings'] = df3['ratings'].astype('int64')
# df.Weight = df.Weight.astype('int64') 


In [ ]:
newdf = out.loc[out['rating'] >= 4.5, :]


In [53]:
df = df.loc[df['ratings'] >= 6, :]

In [54]:
df.describe()

LATITUDE   LONGITUDE     ratings  Language
count  212.000000  212.000000  212.000000     212.0
mean    40.739909  -73.979240    7.128302       0.0
std      0.052709    0.028290    0.657044       0.0
min     40.572232  -74.072721    6.000000       0.0
25%     40.714679  -73.995158    6.600000       0.0
50%     40.752784  -73.981918    7.100000       0.0
75%     40.772313  -73.970931    7.600000       0.0
max     40.859460  -73.841257    9.200000       0.0

In [90]:
df = df.drop(51)

In [55]:
df.to_csv('nyc_movies_for_geojson.csv', index=False)  

In [59]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 233 entries, 0 to 232
Data columns (total 23 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Film                               233 non-null    object 
 1   Year                               233 non-null    object 
 2   URL Encoded name                   233 non-null    object 
 3   Image File Name                    182 non-null    object 
 4   Agency Credit                      183 non-null    object 
 5   Artist Credit                      233 non-null    object 
 6   Director/Filmmaker Name            233 non-null    object 
 7   Director/Filmmaker IMDB Link       233 non-null    object 
 8   Location Display Text              233 non-null    object 
 9   LATITUDE                           233 non-null    float64
 10  LONGITUDE                          233 non-null    float64
 11  Borough                            233 non-null    object 

In [68]:
foo = df['ratings'][0]

In [69]:
type(foo)

str

In [70]:
len(foo)

6

In [71]:
if len(foo) > 1:
    foo = foo[:-3]
print(foo)

6.7


True


In [42]:

foo = foo[:-3]
print(foo)

he
